From 27e91d562224b03e5b18e2f33b1a1800e3477cc3 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Fri, 28 Jul 2023 12:28:44 +0200
Subject: [PATCH 0486/1141] drivers: crypto: stm32-hash: fix algorithms support

Set HASH_CSR_NB_SHA256 to 38 instead of 22 to be compliant with the
reference manual.

Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ib02875b082547ed44882101735aa595044726fef
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/320609
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/crypto/stm32/stm32-hash.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/stm32/stm32-hash.c b/drivers/crypto/stm32/stm32-hash.c
index 0d7d71b6585d..db764e4848a9 100644
--- a/drivers/crypto/stm32/stm32-hash.c
+++ b/drivers/crypto/stm32/stm32-hash.c
@@ -74,7 +74,7 @@
 
 /* Context swap register */
 #define HASH_CSR_NB_SHA256_HMAC		54
-#define HASH_CSR_NB_SHA256		22
+#define HASH_CSR_NB_SHA256		38
 #define HASH_CSR_NB_SHA512_HMAC		103
 #define HASH_CSR_NB_SHA512		91
 #define HASH_CSR_NB_SHA3_HMAC		88
-- 
2.39.2

